File: app/src/main/java/com/jay/currencyconverter/di/BankActivityComponent.java
package com.jay.currencyconverter.di;

import com.jay.currencyconverter.ui.bankActivity.BankActivity;
import com.jay.currencyconverter.ui.bankActivity.IBankView;

import dagger.BindsInstance;
import dagger.Component;

@Component(modules = {BankActivityModule.class})
public interface BankActivityComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder bankPresenter(IBankView view);
        BankActivityComponent build();
    }

    void inject(BankActivity activity);
}

File: app/src/main/java/com/jay/currencyconverter/di/BankActivityComponent.kt
package com.jay.currencyconverter.di

import androidx.fragment.app.FragmentActivity
import com.jay.currencyconverter.ui.organizationActivity.OrganizationActivity
import dagger.BindsInstance
import dagger.Component

@Component(modules = [BankActivityModule::class])
interface BankActivityComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun activity(activity: FragmentActivity) : Builder
        fun build(): BankActivityComponent
    }

    fun inject(activity: OrganizationActivity)
}